requirements-osbs-client.in

```diff
@@ -0,0 +1,2 @@
+https://github.com/containerbuildsystem/osbs-client/archive/refs/tags/1.13.0.tar.gz
```

Comment:
osbs-client is a normal runtime requirement, it should be in requirements.in. I know that will break setup.py, because setup.py is reading requirements from the requirements file, but that shouldn't be a thing in the first place (https://packaging.python.org/discussions/install-requires-vs-requirements/)

- setup.py `install_requires` should be the minimal set of requirements to run the application with the loosest possible version constraints, and should not be read from any requirements file
- requirements.in should be input for pip-compile; it will mostly be a copy of `install_requires` but there may be exceptions (and osbs-client will be one of them)
- requirements.txt should be the set of all recursive dependencies, pinned to exact versions


So in setup.py `install_requires`, we want `osbs-client >= 1.0.0` (well, `2.0.0` for OSBS 2.0, but that doesn't exist yet) and in requirements.in, we want the osbs-client url.

Yes, when adding deps, we will need to add them to both setup.py and requirements.in, but that is _The Way_ (and also how [Cachito does it](https://github.com/release-engineering/cachito/#adding-dependencies))